com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -92,43 +89,38 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
         return StepPhase.POLL;
     }
 
-    private ApplicationResources getApplicationResources(ExecutionWrapper execution, String appArchiveId, String fileName)
-        throws FileStorageException {
-        ApplicationResources applicationResources = new ApplicationResources();
+    private String getNewApplicationDigest(ExecutionWrapper execution, String appArchiveId, String fileName) throws FileStorageException {

```

Comment:
My idea for double reading was to calculate file resources digests and check for already uploaded files and in the second iteration to save files which are missing in the cache. Now double reading is not needed and calculation of digest of the whole content could be done with one reading while writing to zip file. If we keep this behaviour we should add logic to finish UploadAppStep if both digests are equal and skip uploading to the controller.